clients/go/pkg/worker/jobWorker_builder.go

```diff
@@ -190,7 +193,8 @@ func (builder *JobWorkerBuilder) Open() JobWorker {
 	}
 }
 
-func NewJobWorkerBuilder(gatewayClient pb.GatewayClient, jobClient JobClient) JobWorkerBuilderStep1 {
+// NewJobWorkerBuilder should use the same retryPredicate used by the CredentialProvider (ShouldRetry method)
```

Comment:
Can we give an example usage here?

Reply:
In a comment? It's just going to look like straight text, not very readable. Ideally, we'd have example files which help our users see sample usages (good thing to add to the to do list actually)

Reply:
I've seen other Go packages put examples in comments. It looks OK-ish, we just keep it short.

See for example: https://godoc.org/github.com/aws/aws-sdk-go/aws#Config

```go
    // Retryer guides how HTTP requests should be retried in case of
    // recoverable failures.
    //
    // When nil or the value does not implement the request.Retryer interface,
    // the client.DefaultRetryer will be used.
    //
    // When both Retryer and MaxRetries are non-nil, the former is used and
    // the latter ignored.
    //
    // To set the Retryer field in a type-safe manner and with chaining, use
    // the request.WithRetryer helper function:
    //
    //   cfg := request.WithRetryer(aws.NewConfig(), myRetryer)
    //
    Retryer RequestRetryer
```